src/main/java/entity/Deck.java

```diff
@@ -70,4 +78,42 @@ public String getName() {
     public String getId() {
         return id;
     }
+
+    public Deck clone() {
+        Deck result = new Deck(name, id, author);
+
+        for (Card card : deck) {
+            result.addCard(card);
+        }
+
+        return result;
+    }
+
+    public Collection<String> getProblems() {
```

Comment:
Good naming conventions are used here, why is it that you chose the maximum deck length to be 60 cards?